### ISSUE TITLE
Windows 10 Desktop App not launching offline

### ISSUE THREAD

**Issue by Caldefredo** [standardnotes/forum#359]
In attempting to open windows desktop app without an internet connection, all I get is a blank white box. Won't open all the way.

**Comment by mobitar** [comment]
The entire app is white, or just the editor section? Could you do Help -> Toggle Error Console, and see what it displays on launch?

**Comment by Caldefredo** [comment]
The entire app was white. When I got back online yesterday I reinstalled the app. Just tested it. Now the app won't even launch unless I'm online.

**Comment by mobitar** [comment]
Can you perform the error console steps mentioned so we can see why your
app isn’t working?

On Oct 29, 2018, at 5:14 PM, Caldefredo <notifications@github.com> wrote:

The entire app was white. When I got back online yesterday I reinstalled
the app. Just tested it. Now the app won't even launch unless I'm online.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/standardnotes/forum/issues/359#issuecomment-434100786>,
or mute the thread

**Comment by Caldefredo** [comment]
I'm not sure which info you're looking for here, but here's what comes up under the "Console" tab:
C:\Users\[xxxx]\AppData\Local\Programs\standard-notes\resources\app.asar\dist\javascripts\render\spe…:71 Using OS-level spell check API with locale en_US

**Comment by mobitar** [comment]
That's all that comes up? Strange. What version of the app is this? Version 1.x used to be online only, but that was a very long time ago. Any chance you're using a really old version?

**Comment by mobitar** [comment]
Oh, so that output is from a functioning instance? What we'd need is to see what the console shows when the app doesn't work. Hoping to see some clue as to why this is happening, as I can't replicate your issue on my end.

**Comment by Caldefredo** [comment]
Given that it doesn't launch at all, I'm not sure how I would access that.

**Comment by mobitar** [comment]
You mentioned that the app launches, but it's all white? In that case, the console would still show. It's in the same window, but in a separate part of the app that isn't affected by display issues.

**Comment by Caldefredo** [comment]
I installed the recent update, so you can close this. I will let you know of any future problems.

**Comment by nsclong** [comment]
I have the problem on all Windows 10 installations, running 3.0.2 version. When the computers are offline, the application does not 'start' (i.e. the Window with the editor does not open) - I can see the process is opened in the background but the app doesn't open for use.

**Comment by mobitar** [comment]
So no window opens up at all? Can you try restarting your computer and see if that makes a difference?

**Comment by nsclong** [comment]
That is correct. And this happens on all of my computers: 2 laptops and one desktop. I have tried restarting them.

**Comment by nsclong** [comment]
Hi Mo (@mobitar), I have upgraded to 3.0.3 - I am a paid up member of Extensions. But still I have the same issue of being unable to use Standard Notes offline.

**Comment by mobitar** [comment]
The issue is that I simply cannot replicate this. I'm testing on Windows 10 version 1709. I turn off Wifi, and disconnect the ethernet cable. I launch Standard Notes. It opens as usual. I can't fix the issue unless I can replicate it. If you can try to tinker around to see what kind of variables lead to this happening, that would be helpful. If you can test this out when you actually have a working connection, that would be good. That is, have a good connection, open the app, confirm it works. Then, disconnect your connection, and see what happens.

**Comment by nsclong** [comment]
Hi Mo, thanks for trying to help fix this. I am on Windows 10.0.17763. I have tried opening SN with internet: it works. Then close the app, disconnect internet and it will not open. This is the same for all three machines I have, so if it's just me with this problem it suggests that it is something to do with my Standard Notes / Extensions configuration that is being shared via the cloud server. The only thing I can think of that would make me 'different' is that in a previous SN update, my SN Extensions got out of sync and I had to uninstall and reinstall various extensions (including Batch Manager). They appear to all work (when connected to internet) and be syncing correctly now, but perhaps some configuration got broken. Perhaps the answer is for me to uninstall and reinstall all machines again? But I need to be absolutely sure I don't lose my data. Is there a sure way to uninstall so that there is no trace left on each machine, before re-installing again?

**Comment by mobitar** [comment]
You won't lose any data if it's synced to your account. The way to verify this is to sign in to the web app at app.standardnotes.org, and make sure that all your data looks synced. If so, it's safe to uninstall the app and reinstall it, then sign back in to your account.

If it still doesn't work, please do this in offline mode so we can see the live log of what the application is reporting:

1. Close Standard Notes.
2. Open the Command Prompt application.
3. Paste `cmd /K "C:\Users\UserName\AppData\Local\Programs\standard-notes\Standard Notes.exe"`, but replace `UserName` with whatever your current Windows account is.
4. Press enter, and the application should begin. Please share what the Command Prompt window logs. There's likely some kind of error occurring.

**Comment by nsclong** [comment]
When I do the command prompt as you specify, when online it works without error, but when I go into flight mode with no internet the command hangs - I get no error messages, no app window, and no command prompt back.

**Comment by nsclong** [comment]
Bandwidth problem is resolved this morning, so I downloaded version 3.0.3 from the SN website.

On my laptop I uninstalled Standard Notes and deleted the Standard Notes folder that was left in AppData/Roaming after the uninstall - so hopefully all traces of the app removed.
I then reinstalled from the download file.
**I have not logged into my account at all for any of the following** - so this is a fresh install with none of my details (unless there are some left in the Windows registry or somewhere I don't know about).

When I started Standard Notes with an internet connection it opened fine.

I restarted the computer and put it into flight mode. Then when I start SN, the app just hangs. However when I go to Windows Task Manager I see there are two SN processes running in the background - see the screenshot.
![windowstaskmanager](https://user-images.githubusercontent.com/16524531/52108834-42d44900-25f3-11e9-8e8a-350a693abe95.png)

If I then reconnect internet and start SN, I get a popup error as follows (I guess this just confirms that those processes are already running in the background).
![error popup](https://user-images.githubusercontent.com/16524531/52108975-c726cc00-25f3-11e9-9303-fcf8501cf567.png)

I am running Windows 10 Home v 10.0.17763 and have no anti-virus programs.  I have the same problem on all my other computers.

As I stated previously, there are no error messages given when I start the app from the Command Prompt.

**Comment by mobitar** [comment]
Hmm..very strange. Can you try this:

When in offline mode and you see two process, kill both processes. Then try launching Standard Notes again. Does it launch? If not, how many processes do you see for it?

**Comment by nsclong** [comment]
The first time I start it, it only starts one process.
Then if I kill that process and start it again, it starts two.
If I kill the two processes and start it again, it starts two again.

**Comment by mobitar** [comment]
This is offline? And in that first time when it starts one process, does it show a window or no? Have you tried yet killing all processes then launching the app via the Command Prompt line?

**Comment by nsclong** [comment]
Yes - this is offline. When online it works fine.
The first time with one process, there is no window (in fact even after restarting I am always getting 2 processes now).
When I kill the processes and start with **cmd /K** it just hangs (no window, no errors).

**Comment by mobitar** [comment]
That's just so strange. I don't have any leads as to why that would happen. My only thought is perhaps it's the autoupdate script trying to access the network but then fails and hangs the application? It should fail gracefully, and does in my case. What we could do next is find out what network requests the app tries to make at launch. We'd have to use a third party tool. I don't know of any offhand, but the goal is to find some kind of network monitoring or proxy tool for Windows so we can see all network requests that are going through.

Again, I can't replicate this issue as hard as I try, so the fate of this issue depends on you :) Sorry to throw this at you. Feel free to do some googling for a tool as such, and if not, I can do some research and suggest one. Not too sure what you would rate your tech savviness level as..

**Comment by nsclong** [comment]
Hi Mo, I appreciate your continuing follow up, as I love Standard Notes, but this is a bit of a show stopper for me. 
I think I have a clue that might help. I also run VS Code on my machine and have discovered it has the same problem. I believe that both VS Code and SN share Electron as their underlying platform, so I conclude the fault may be with Electron. I have found the following issue regarding VS Code, and other users have recently experienced the same issue as me: https://github.com/Microsoft/vscode/issues/7570  This issue suggests that it may be fixed in the next release of VS Code, but it does not reference the underlying Electron problem. The work around suggested in the meantime is to use a loopback adaptor, so I think you are on the right track thinking it's network related. I will continue to search for the underlying Electron issue, and if I find it let you know. If you find it first please let me know (or perhaps provide a fix in the next version) as I am sure I will not be the only person affected by this issue.
In answer to your question about my technical abilities.... I code, but am no network expert.

**Comment by mobitar** [comment]
Ah interesting. Nice finds. Issue dates back to 2016 surprisingly. Although, [this comment](https://github.com/Microsoft/vscode/issues/7570#issuecomment-460235178) was just added 6 hours ago:

```
Last week i was looking what is eating up all my ssd space and found an directory from code.
VSCode wasnt working anyway so ive deleted anything in these Folders.

C:\Users\xxx\AppData\Roaming\Code\Cache
C:\Users\xxx\AppData\Roaming\Code\CachedData
C:\Users\xxx\AppData\Roaming\Code\CachedExtensions

After starting vscode by mistake and not insiders it started fine without Internet.
```

Can you try deleting the same directories, but for Standard Notes? (Ensure your data is properly synced before doing so, in case this has unwanted side effects.)

**Comment by nsclong** [comment]
I tried deleting those three folders. Unfortunately I still have the same issue.

**Comment by mobitar** [comment]
I'm afraid I don't know what else to try. We'll have to monitor those two issues to see if anyone finds another resolution, or at least wait for someone else to have the same issue with SN you're having, so that we have a wider range of replication. Sorry :/

**Comment by mobitar** [comment]
I'll take a look at upgrading to v3 or v4 electron in the next release. Have to make sure nothing breaks, as updates like these tend to make large changes to underlying fundamentals.

**Comment by nsclong** [comment]
That would be really great Mo, and I would be much obliged.

**Comment by FelixWeigand** [comment]
Hello there, I also got the problem that I can't use Standard Notes without internet on Windows 10. If I pull the ethernet cable before starting the desktop version it will not let me log in ("Server error"). If I'm already logged in and pull the cable, I'm still able to read notes, but can't create new views or notes. Have you got some new information on the "Windows 10 offline issue"..? Thanks & cheers, Felix

**Comment by FelixWeigand** [comment]
... I set a local passcode as recommended in your Help sites, but that doesn't change anything..

**Comment by mobitar** [comment]
You should definitely be able to launch and view your existing notes offline. You can also create new notes offline, but they won't sync to your other devices until you're back online. Can you describe what you're seeing in detail when you're offline?

**Comment by FelixWeigand** [comment]
Thanks for your quick reply! Here we go:
- Launching the programme and viewing the existing notes works well. I see the normal web surface of your tool. 
- When I try to create a new view, it creates the view and I am able to name it - however, when I then try to give the associated first note a title it only let's me type a few letters before an error message appears ("There was an error saving your note. Please try again."); also, the content field doesn't react if I click into it to type something. 
- After that, I'm not able to create new views anymore, but creating notes within an existing view will still work for one or two notes; after that the title and content area of all notes won't react anymore (I can still create new notes, but not edit title or content). 
- Another message comes up when I try to edit the notes in vain ("Please use FileSafe or the bold editor to attach images and files."), though I haven't tried to attach anything. 
- On the top left, it says "Sync Unreachable. Changes saved offline." - That would bee fine if any changes were possible..

**Comment by mobitar** [comment]
Very strange. Thanks for the detailed steps. It might be something with our 3.0.16 release. Will need to investigate.

**Comment by Loxxlie** [comment]
Just to add to this, I'm seeing a similar issue after installing Standard Notes on my Windows Surface Go. It often occurs when I lose connectivity. The App looks fine upon startup but after tying a few words or waiting a few seconds I get a "Sync Unreachable" error and everything becomes uneditable. I'll try and record a video later and follow some of the remediation suggestions listed here.

**Comment by effieeee** [comment]
No similar issues on latest version so far

**Comment by iso1048** [comment]
I'm using the Windows Desktop app and have been encountering the same problem as of today, where the app window opens but only a white screen is shown (regardless of whether or not I'm online). Everything works fine when I sign in to the online web app. I've attached an image of what the console shows. I also installed the latest version (3.181.31) but the problem stil persists. I'm hoping that I won't have to find an alternative app.

Any help would be much appreciated. Thanks. 

<img width="830" alt="error" src="https://github.com/standardnotes/forum/assets/22124489/8ed6c29e-fa48-4097-91bd-a3f6a27c2888">

**Comment by amanharwara** [comment]
@iso1048 We're working on a fix for this issue, until then please downgrade to https://github.com/standardnotes/app/releases/tag/%40standardnotes%2Fdesktop%403.181.29

**Comment by iso1048** [comment]
> @iso1048 We're working on a fix for this issue, until then please downgrade to https://github.com/standardnotes/app/releases/tag/%40standardnotes%2Fdesktop%403.181.29

Thank you, appreciate the help!

**Comment by effieeee** [comment]
Hi @iso1048, the fix has now been pushed on v3.181.32. Apologies for the trouble this caused! 🙏

https://github.com/standardnotes/app/releases/tag/%40standardnotes%2Fdesktop%403.181.32
